library/axi_dmac/bd/bd.tcl

```diff
@@ -22,14 +22,19 @@ proc init {cellpath otherInfo} {
 	}
 
 	foreach dir {SRC DEST} {
-		# This is a bit of a hack, but we can't change the protocol if the type
-		# is not AXI MM
+		# Change the protocol by first enabling the parameter - setting the type to AXI MM
 		set old [get_property "CONFIG.DMA_TYPE_${dir}" $ip]
 		set_property "CONFIG.DMA_TYPE_${dir}" "0" $ip
 		set_property "CONFIG.DMA_AXI_PROTOCOL_${dir}" $axi_protocol $ip
 		set_property "CONFIG.DMA_TYPE_${dir}" $old $ip
 	}
 
+	# Change the protocol by first enabling the parameter - enabling the SG transfers 
+	set old [get_property "CONFIG.DMA_SG_TRANSFER" $ip]
```

Comment:
Because the Scatter-Gather interface has only one type of AXI interface (AXI-MM), unlike the Source interface, which has 3 types selectable through `DMA_TYPE_SRC`. The parameter is similar to `DMA_2D_TRANSFER`.